app/views/spree/layouts/admin.html.haml

```diff
@@ -4,80 +4,12 @@
-
-      %header#header{"data-hook" => ""}
-        .container
-          %figure.columns.five{"data-hook" => "logo-wrapper"}
-            = link_to image_tag(Spree::Config[:admin_interface_logo], :id => 'logo'), spree.admin_path
-          %nav.columns.eleven{"data-hook" => "admin_login_navigation_bar"}
-
-      %nav#admin-menu{"data-hook" => ""}
-        .container
-          .sixteen.columns.main-menu-wrapper
-            %ul.inline-menu.fullwidth-menu{"data-hook" => "admin_tabs"}
-              = render :partial => 'spree/admin/shared/tabs'
-
-      - if content_for?(:sub_menu)
-        %nav#sub-menu{"data-hook" => ""}
-          .container
-            .sixteen.columns
-              = yield :sub_menu
-
-      - if content_for?(:page_title) || content_for?(:page_actions)
-        #content-header{"data-hook" => ""}
-          .container
-            .sixteen.columns
-              .block-table
-                - if content_for?(:page_title)
-                  .table-cell
-                    %h1{:class => "page-title"}= yield :page_title
-                - if content_for?(:page_actions)
-                  .page-actions.table-cell.toolbar{"data-hook" => "toolbar"}
-                    %ul.inline-menu
-                      = yield :page_actions
-
-      .container
-        .row
-          - content_class = content_for?(:sidebar) ? "with-sidebar" : ""
-          #content{:class => content_class, "data-hook" => ""}
-            - if content_for?(:table_filter)
-              - table_filter_class = content_for?(:sidebar) ? 'twelve columns' : 'sixteen columns'
-              #table-filter{:class => table_filter_class, "data-hook" => ""}
-                %fieldset
-                  %legend{:align => "center"}= yield :table_filter_title
-                  = yield :table_filter
-            - div_class = content_for?(:sidebar) ? 'twelve columns' : 'sixteen columns'
-            %div{:class => div_class}
-              = yield
-            - if content_for?(:sidebar)
-              %aside#sidebar.four.columns{"data-hook" => ""}
-                - if content_for?(:sidebar_title)
-                  %h5.sidebar-title
-                    %span= yield :sidebar_title
-                = yield :sidebar
-
-    %div{"data-hook" => "admin_footer_scripts"}
-
-    = render 'spree/shared/google_analytics'
-
-    %script
-      = raw "Spree.api_key = \"#{try_spree_current_user.try(:spree_api_key).to_s}\";"
+    - if content_for?(:main_ng_app_name)
+      - if content_for?(:main_ng_ctrl_name)
+        %div{ "ng-app" => yield(:main_ng_app_name).strip.html_safe, "ng-controller" => yield(:main_ng_ctrl_name).strip.html_safe }
+          = render :partial => 'spree/layouts/admin_body'
+      - else
+        %div{ "ng-app" => yield(:main_ng_app_name).strip.html_safe }
+          = render :partial => 'spree/layouts/admin_body'
+    - else
+      = render :partial => 'spree/layouts/admin_body'
```

Comment:
I tried a few times and failed to make this look better...

Reply:
I think is the least of the problems. It's a negligible downside for an incredible upside this PR brings